OCC.Service.Webhost/Repositories/PersonRepository.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using OCC.Data;
+using OCC.Service.Webhost.Services;
+using OCC.Service.Webhost.Tools;
+using Person = OCC.Service.Webhost.Services.Person;
+
+namespace OCC.Service.Webhost.Repositories
+{
+    public class PersonRepository
+    {
+        private readonly OCCDB _dbContext;
+
+        public PersonRepository(OCCDB dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public int RegisterPerson(Person person)
+        {
+            var p = new OCC.Data.Person();
+            Mapper.CopyProperties(person, p);
+
+            _dbContext.People.Add(p);
+            _dbContext.SaveChanges();
+
+            return p.ID;
+        }
+
+        public Person Login(Person person)
+        {
+            Person dcAttendee = default(Person);
+
+            var bcAttendee =
+                _dbContext.People
+                    .SingleOrDefault(p =>
+                        p.Email == person.Email &&
+                        p.PasswordHash == person.PasswordHash);
+
+            if (bcAttendee != null)
+            {
+                dcAttendee = new Person();
+                Mapper.CopyProperties(bcAttendee, dcAttendee);
+            }
+
+            return dcAttendee;
+        }
+
+        public Person FindPersonByEmail(string email)
+        {
+            Person dcAttendee = default(Person);
+
+            var bcAttendee = _dbContext.People.Where(p => p.Email == email)
+                .SingleOrDefault();
+
+            if (bcAttendee != null)
+            {
+                dcAttendee = new Person();
+                Mapper.CopyProperties(bcAttendee, dcAttendee);
+            }
+            return dcAttendee;
+        }
+
+        public void ResetPassword(string emailAddress, string temporaryPassword, string temporaryPasswordHash)
+        {
+            if (String.IsNullOrEmpty(emailAddress))
+            {
+                throw new ArgumentNullException(emailAddress, "email address must be provided.");
+            }
+
+            var bcAttendee =
+                _dbContext.People.SingleOrDefault(p => p.Email == emailAddress);
+
+            if (bcAttendee == null)
+            {
+                throw new ArgumentOutOfRangeException(emailAddress, "attendee was not found.");
+            }
+
+            if (!String.IsNullOrEmpty(temporaryPasswordHash))
+            {
+                bcAttendee.PasswordHash = temporaryPasswordHash;
+            }
+            _dbContext.SaveChanges();
+
+            IMailService svc = new SmtpMailService();
```

Comment:
Since we are using DI, it would make sense to inject SmtpMailService instead of instantiating it here.